src/handlers/assign.rs

```diff
@@ -799,44 +799,86 @@ async fn find_reviewer_from_names(
         .to_string())
 }
 
+#[derive(Eq, PartialEq, Hash)]
+enum ReviewerCandidate {
+    /// This reviewer was directly requested for a review.
+    Direct(String),
+    /// This reviewer was expanded from a team or an assign group.
+    Expanded(String),
+}
+
+impl AsRef<str> for ReviewerCandidate {
+    fn as_ref(&self) -> &str {
+        match self {
+            ReviewerCandidate::Direct(name) => name.as_str(),
+            ReviewerCandidate::Expanded(name) => name.as_str(),
+        }
+    }
+}
+
+impl ReviewerCandidate {
+    fn into_name(self) -> String {
+        match self {
+            ReviewerCandidate::Direct(name) => name,
+            ReviewerCandidate::Expanded(name) => name,
+        }
+    }
+}
+
 /// Recursively expand all teams and adhoc groups found within `names`.
 /// Returns a set of expanded usernames.
 /// Also normalizes usernames from `@user` to `user`.
-///
-/// Returns `(set of expanded users, expansion_happened)`.
-/// `expansion_happened` signals if any expansion has been performed.
 fn expand_teams_and_groups(
     teams: &Teams,
     issue: &Issue,
     config: &AssignConfig,
     names: &[String],
-) -> Result<(HashSet<String>, bool), FindReviewerError> {
-    let mut expanded = HashSet::new();
-    let mut expansion_happened = false;
+) -> Result<HashSet<ReviewerCandidate>, FindReviewerError> {
+    let mut selected_candidates: HashSet<String> = HashSet::new();
 
     // Keep track of groups seen to avoid cycles and avoid expanding the same
     // team multiple times.
     let mut seen_names = HashSet::new();
 
+    enum Candidate<'a> {
+        Direct(&'a str),
+        Expanded(&'a str),
+    }
```

Comment:
Can we re-use `ReviewerCandidate`? Maybe by having a generic `S = String` that would be `&str` here and stay the default elsewhere.

Reply:
Sure, no problem.